### ISSUE TITLE
Add a Formant Filter

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#1584]
Chatting with @JackyLigon on slack this morning and he suggested a new filter type, the Formant filter.

I said 

>And Formant filter - you mean just ‘3 poles medium-tight set of bandpass in one filter spread to human sounds’ type thing? Or something more than that

And he said “yes”

And then I suggested we would code up this table

![](https://www.phon.ucl.ac.uk/courses/plin0065/images/vowel-formant-table.png)

From https://www.phon.ucl.ac.uk/courses/plin0065/week5.php and he said “yup”

Then we concluded

> The DSP for that is super easy to construct in surge. Need to think about how it would fit into the filter banks. Presumably we would want controls for where in Formant space you were though so more than just cutoff and resonance. Something like ‘vowel sound’ and ‘base frequency’ and ‘resonance’ seems to be what you would want

> In theory my skinning engine in 1.7 lets us add a parameter. So we might be able to do it. It is a nice small unambiguous change for 1.7 that proves our point too

And ended with 

> As we know, routing a noise component, as well as the oscillators, through a format filter, can produce some quite extraordinary timbres of a vocal tract nature.

**Comment by mkruselj** [comment]
One thing that would make this more usable is being able to choose between which vowels you really want to morph.

Just going through a list of vowel sounds as in that table is repetitive and boring. But if we could pick, say, 5 different vowels, and have a parameter to morph through them, then we're in for a lot more fun!

Of course, then it's not adding just a single parameter, alas...

**Comment by baconpaul** [comment]
most of the research seems to show the vowels laid out in an x y plane with 'dead spots'. probably some idea there.

**Comment by mkruselj** [comment]
Here's a brilliantly executed formant filter:

https://youtu.be/rP7ILrztu80?t=90

Important stuff starts at 1:30 onward.

**Comment by baconpaul** [comment]
true. but then it's not as fun since you can't do polyphonic framptons

**Comment by mkruselj** [comment]
We could borrow MorphEQ from Shortcircuit here. It's bog-standard biquads, but maybe we could use something a bit nicer/more modern (not necessarily non-linear, but perhaps ZDF) with @jatinchowdhury18's help. 🙂
